projects/netgrif-components-core/src/lib/navigation/service/uri.service.ts

```diff
@@ -1,34 +1,39 @@
-import {Injectable, OnDestroy} from '@angular/core';
-import {LoggerService} from '../../logger/services/logger.service';
-import {UriResourceService} from './uri-resource.service';
+import {HttpParams} from '@angular/common/http';
+import {Inject, Injectable, OnDestroy, Optional} from '@angular/core';
 import {BehaviorSubject, Observable, of} from 'rxjs';
-import {UriNodeResource} from '../model/uri-resource';
-import {Case} from '../../resources/interface/case';
-import {CaseResourceService} from '../../resources/engine-endpoint/case-resource.service';
-import {SimpleFilter} from '../../filter/models/simple-filter';
+import {map} from 'rxjs/operators';
 import {CaseSearchRequestBody, PetriNetSearchRequest} from '../../filter/models/case-search-request-body';
+import {SimpleFilter} from '../../filter/models/simple-filter';
 import {ActiveGroupService} from '../../groups/services/active-group.service';
-import {map} from "rxjs/operators";
-import {LoadingEmitter} from "../../utility/loading-emitter";
+import {LoggerService} from '../../logger/services/logger.service';
+import {CaseResourceService} from '../../resources/engine-endpoint/case-resource.service';
+import {Case} from '../../resources/interface/case';
+import {Page} from '../../resources/interface/page';
+import {LoadingEmitter} from '../../utility/loading-emitter';
+import {PaginationParams} from '../../utility/pagination/pagination-params';
+import {NAE_URI_NODE_CASES_PAGE_SIZE} from '../model/size-menu-injection-token';
+import {UriNodeResource} from '../model/uri-resource';
+import {UriResourceService} from './uri-resource.service';
 
 /**
  * Service for managing URIs
  * */
 @Injectable({
-    providedIn: 'root'
+    providedIn: 'root',
 })
 export class UriService implements OnDestroy {
 
     private static ROOT: string = 'root';
     private _rootNode: UriNodeResource;
     private readonly _rootLoading$: LoadingEmitter;
     private readonly _parentLoading$: LoadingEmitter;
-    private readonly _activeNode$: BehaviorSubject<UriNodeResource>
+    private readonly _activeNode$: BehaviorSubject<UriNodeResource>;
 
     constructor(protected _logger: LoggerService,
                 protected _resourceService: UriResourceService,
                 protected _caseResourceService: CaseResourceService,
-                protected _activeGroupService: ActiveGroupService) {
+                protected _activeGroupService: ActiveGroupService,
+                @Optional() @Inject(NAE_URI_NODE_CASES_PAGE_SIZE) protected pageSize: string = '20') {
```

Comment:
the default value will never be applied, since `@Optional()` injects `null` if the token is not provided, and default values are only applied to `undefined` parameters

Reply:
Thanks for the warning. It is fixed inside the constructor.